### PR TITLE
Fix: Metric format on serialized metric mode

### PR DESCRIPTION
**Pull request by Xuefeng-Zhu**
The default name of metric column is not displayed correctly for metric with parameters

### CODE REVIEW

**Comment by Xuefeng-Zhu** [comment]
@mattgraham1995 updated

File: packages/data/addon/helpers/metric-format.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ import { hasParameters } from '../utils/metric';
 import { get } from '@ember/object';
 import { isPresent } from '@ember/utils';
 import Helper from '@ember/component/helper';
+import isObject from 'lodash/isObject';
```
Comment (mattgraham1995) [inline packages/data/addon/helpers/metric-format.js]: Do we need this or can we just use `typeof val === 'object'`

Comment (cythrawll) [reply packages/data/addon/helpers/metric-format.js]: `typeof val === 'object'` is pretty nasty if you get nulls and other objects that don't work like the plain ol js objects.

Comment (mattgraham1995) [reply packages/data/addon/helpers/metric-format.js]: or maybe `val === Object(val)` ?

Comment (Xuefeng-Zhu) [reply packages/data/addon/helpers/metric-format.js]: Still prefer _.isObject, since it is easier to read and understand

Comment (Xuefeng-Zhu) [reply packages/data/addon/helpers/metric-format.js]: @mattgraham1995 Actually need to use `isPlainObject` to exclude array case, so `val === Object(val)  ` will not work here

File: packages/data/tests/unit/helpers/metric-format-test.js [hunk 2]
```diff
@@ -17,6 +17,9 @@ test('It formats metrics appropriately', function(assert) {
   assert.equal(metricFormat({metric: {name: 'foo', longName: 'Foo'}, parameters: {p: 1}}, 'Bar'),
     'Bar (1)',
     'Formats metric with parameters');
+  assert.equal(metricFormat({ metric: { name: 'foo', longName: 'Foo' }, parameters: { p: { defaultValue:1 } } }, 'Bar'),
+    'Bar (1)',
+    'Formats metric with parameters');
```
Comment (jkusa) [inline packages/data/tests/unit/helpers/metric-format-test.js]: @Xuefeng-Zhu Can we update this msg to be more specific than the msg above?

Comment (Xuefeng-Zhu) [reply packages/data/tests/unit/helpers/metric-format-test.js]: updated

File: packages/data/tests/unit/helpers/metric-format-test.js [hunk 3]
```diff
@@ -17,6 +17,9 @@ test('It formats metrics appropriately', function(assert) {
   assert.equal(metricFormat({metric: {name: 'foo', longName: 'Foo'}, parameters: {p: 1}}, 'Bar'),
     'Bar (1)',
     'Formats metric with parameters');
+  assert.equal(metricFormat({ metric: { name: 'foo', longName: 'Foo' }, parameters: { p: { defaultValue:1 } } }, 'Bar'),
```
Comment (jkusa) [inline packages/data/tests/unit/helpers/metric-format-test.js]: Are we sure about this object shape?
```js
{ metric: { name: 'foo', longName: 'Foo' }, parameters: { p: { defaultValue:1 } } }
```

Also, can we verify the fix with an acceptance test?

Comment (cythrawll) [reply packages/data/tests/unit/helpers/metric-format-test.js]: So this shape does look correct.  However I still need to figure out how to produce this for an acceptance test.

**Comment by cythrawll** [comment]
acceptance test added!

File: packages/reports/tests/acceptance/navi-report-test.js [hunk 4]
```diff
@@ -1709,4 +1709,16 @@ test("adding metrics to reordered table keeps order", function(assert) {
       'Total Clicks'
     ], 'The headers are reordered as specified by the reorder');
   });
+});
+
+test('Paramaterized metrics with default displayname are not considered custom', function(assert) {
```
Comment (mattgraham1995) [inline packages/reports/tests/acceptance/navi-report-test.js]: Parameterized

**Pull request merged**